### PR TITLE
enable sticky-note pop-up toggling + annotation type

### PR DESCRIPTION
**Pull request by Maxwell-Yang-PDFTron**
sticky-note pop-up could be disabled by overrideBehavior. It would call onBehaviorActivated if selected.

Annotation type is now added as part of the promise result for onAnnotationsSelected, onAnnotationChanged and onQuickMenuButtonPressed.

### CODE REVIEW

**Review by sgong-pdftron — APPROVED** [review]

File: ios/RNTPTDocumentView.m [hunk 1]
```diff
@@ -2236,48 +2238,68 @@ - (BOOL)rnt_documentViewControllerIsNavigationBarEnabled:(PTDocumentBaseViewCont
     return !self.hideTopAppNavBar;
 }
 
-- (NSArray<NSDictionary<NSString *, id> *> *)annotationDataForAnnotations:(NSArray<PTAnnot *> *)annotations pageNumber:(int)pageNumber pdfViewCtrl:(PTPDFViewCtrl *)pdfViewCtrl
+- (NSDictionary<NSString *, id> *)getAnnotationData:(PTAnnot *)annot pageNumber:(int)pageNumber pdfViewCtrl:(PTPDFViewCtrl *)pdfViewCtrl {
```
Comment (JamesTron) [inline ios/RNTPTDocumentView.m:2241]: Lock needed? Could someone ever use this method outside the "all" loop in the method below?

**Review by JamesTron — APPROVED** [review]
Approve, but one change may or may not be needed.

File: ios/RNTPTDocumentView.m (hunk 1 above)
Comment (Maxwell-Yang-PDFTron) [reply ios/RNTPTDocumentView.m:2241]: Yes lock is needed when calling this function. Should I add a comment before this function, or move the lock from before for loop to this function?

Comment (JamesTron) [reply ios/RNTPTDocumentView.m:2241]: Actually thinking more, I think this is good as is. :)

**Pull request merged**